mypy/types.py

```diff
@@ -437,16 +439,26 @@ def accept(self, visitor: 'TypeVisitor[T]') -> T:
 
     type_ref = None  # type: str
 
-    def serialize(self) -> JsonDict:
+    def serialize(self) -> Union[JsonDict, str]:
+        assert self.type is not None
+        type_ref = self.type.alt_fullname or self.type.fullname()
+        if not self.args:
+            return type_ref
         data = {'.class': 'Instance',
                 }  # type: JsonDict
-        assert self.type is not None
-        data['type_ref'] = self.type.alt_fullname or self.type.fullname()
-        if self.args:
-            data['args'] = [arg.serialize() for arg in self.args]
+        data['type_ref'] = type_ref
+        data['args'] = [arg.serialize() for arg in self.args]
         return data
 
     @classmethod
+    def deserialize_ex(cls, data: Union[JsonDict, str]) -> 'Instance':
```

Comment:
I see no reason to make this a separate function. You can just fold this into Instance.deserialize().